src/main/java/org/broadinstitute/hellbender/tools/spark/sv/discovery/AlignmentInterval.java

```diff
@@ -0,0 +1,220 @@
+package org.broadinstitute.hellbender.tools.spark.sv.discovery;
+
+import com.esotericsoftware.kryo.DefaultSerializer;
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import com.google.common.annotations.VisibleForTesting;
+import htsjdk.samtools.Cigar;
+import htsjdk.samtools.SAMFlag;
+import htsjdk.samtools.SAMRecord;
+import htsjdk.samtools.TextCigarCodec;
+import org.broadinstitute.hellbender.tools.spark.sv.utils.SvCigarUtils;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.bwa.BwaMemAlignment;
+import org.broadinstitute.hellbender.utils.read.CigarUtils;
+
+import java.util.List;
+
+import static org.broadinstitute.hellbender.tools.spark.sv.StructuralVariationDiscoveryArgumentCollection.DiscoverVariantsFromContigsAlignmentsSparkArgumentCollection.MISSING_NM;
+
+/**
+ * Each assembled contig should have at least one such accompanying structure, or 0 when it is unmapped.
+ */
+@DefaultSerializer(AlignmentInterval.Serializer.class)
+public final class AlignmentInterval {
+
+    public final SimpleInterval referenceSpan;
+    public final int startInAssembledContig;   // 1-based, inclusive
+    public final int endInAssembledContig;     // 1-based, inclusive
+
+    public final Cigar cigarAlong5to3DirectionOfContig;
+
+    public final boolean forwardStrand;
+    public final int mapQual;
+    public final int mismatches;
+
+    public final int alnScore;
+    public final boolean isFromSplitGapAlignment;
+
+    @VisibleForTesting
+    public AlignmentInterval(final SAMRecord samRecord) {
+
+        final boolean isMappedReverse = samRecord.getReadNegativeStrandFlag();
+        this.referenceSpan = new SimpleInterval(samRecord);
+        this.startInAssembledContig = getAlignmentStartInOriginalContig(samRecord);
+        this.endInAssembledContig = getAlignmentEndInOriginalContig(samRecord);
+
+        this.cigarAlong5to3DirectionOfContig = isMappedReverse ? CigarUtils.invertCigar(samRecord.getCigar())
+                                                               : samRecord.getCigar();
+        this.forwardStrand = !isMappedReverse;
+        this.mapQual = samRecord.getMappingQuality();
+        final Integer nMismatches = samRecord.getIntegerAttribute("NM");
+        this.mismatches = nMismatches==null ? MISSING_NM : nMismatches;
+
+        int alignerScore;
+        try {
+            alignerScore = samRecord.getIntegerAttribute("AS");
```

Comment:
I tried, but to my surprise there is no such method as `hasAttribute(String)` in `SAMRecord`.